<launch>
  <arg name="depth_topic" default="/camera/aligned_depth_to_color/image_raw" />
  <arg name="rgb_topic" default="/camera/color/image_raw/compressed" />
  <arg name="camera_info_topic" default="/camera/aligned_depth_to_color/camera_info" />
  <arg name="stretch_robot_rotate_image_90deg" default="true" />

  <!-- Contact GraspNet Parameters -->
  <arg name="contact_graspnet_ckpt_dir" default="checkpoints/scene_test_2048_bs3_hor_sigma_001" />
  <arg name="z_min" default="0.2" />
  <arg name="z_max" default="1.8" />
  <arg name="local_regions" default="true" />
  <arg name="filter_grasps" default="true" />
  <arg name="forward_passes" default="1" />
  <arg name="skip_border_objects" default="true" />
  <arg name="segmap_id" default="0" />

  <node
    pkg="contact_graspnet"
    name="inference_node"
    launch-prefix="$(find contact_graspnet)/.venv/bin/python"
    type="python_entrypoint.py"
    cwd="node"
    output="screen"
  >
    <param name="depth_topic" value="$(arg depth_topic)" />
    <param name="rgb_topic" value="$(arg rgb_topic)" />
    <param name="camera_info_topic" value="$(arg camera_info_topic)" />
    <param name="stretch_robot_rotate_image_90deg" value="$(arg stretch_robot_rotate_image_90deg)" />

    <param name="contact_graspnet_ckpt_dir" value="$(arg contact_graspnet_ckpt_dir)" />
    <param name="z_min" value="$(arg z_min)" />
    <param name="z_max" value="$(arg z_max)" />
    <param name="local_regions" value="$(arg local_regions)" />
    <param name="filter_grasps" value="$(arg filter_grasps)" />
    <param name="forward_passes" value="$(arg forward_passes)" />
    <param name="skip_border_objects" value="$(arg skip_border_objects)" />
    <param name="segmap_id" value="$(arg segmap_id)" />
  </node>

</launch>
